### PR TITLE
= =last change= =thanks for your source

### PR DESCRIPTION
**Pull request by TXIuTnVsbA**
Add slow walk(fakewalk?),
more antiaim option;

### CODE REVIEW

File: csgo/features/misc/misc.cpp [hunk 1]
```diff
@@ -465,3 +465,26 @@ void c_misc::no_smoke( ClientFrameStage_t stage ) {
 		mat->SetMaterialVarFlag( MATERIAL_VAR_NO_DRAW, g_vars.visuals.misc.remove_smoke );
 	}
 }
+
+void c_misc::slow_walk(CUserCmd* cmd) {
```
Comment (EternityX) [inline csgo/features/misc/misc.cpp]: This is not a good way to achieve 'slow walk'

File: deadcell/util/KeyState.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#pragma once
```
Comment (EternityX) [inline deadcell/util/KeyState.h]: This file is redundant, use ```
g_input.key_pressed()```

File: csgo/menu/menu.cpp [hunk 3]
```diff
@@ -188,15 +188,18 @@ void c_main_form::ragebot_tab() {
...
+	auto yaw_flag = new c_slider("Yaw_flag", antiaim_page, 0, 90, &g_vars.antiaim.yaw_flag, 10);
+	auto pitch_flag = new c_slider("Pitch_flag", antiaim_page, 0, 40, &g_vars.antiaim.pitch_flag, 10);
+
 
 	g_menu.set_y_pos( 10 );
 	auto fakelag_check = new c_checkbox( "Enabled", fakelag_page, &g_vars.misc.fakelag.enabled );
 	auto fakelag_type = new c_combo( "Type", { "Maximum", "Adaptive" }, fakelag_page, 4, &g_vars.misc.fakelag.type, antiaim_groupbox->GetWidth() - 15 );
 	auto fakelag_on_jump_check = new c_combo( "Air", { "Default", "On land", "On jump" }, fakelag_page, 3, &g_vars.misc.fakelag.lagjump, antiaim_groupbox->GetWidth() - 15 );
-	auto fakelag_choke_check = new c_slider( "Choke", fakelag_page, 0, 14, &g_vars.misc.fakelag.amount, 0 );
+	auto fakelag_choke_check = new c_slider( "Choke", fakelag_page, 0, 30, &g_vars.misc.fakelag.amount, 6 );
```
Comment (EternityX) [inline csgo/menu/menu.cpp]: You can't choke 30 ticks.

File: csgo/features/vars.h [hunk 4]
```diff
@@ -100,6 +100,8 @@ class vars {
 			int amount = 6;
 			int lagjump;
 		} fakelag;
+		bool slowWalk = true;
```
Comment (EternityX) [inline csgo/features/vars.h]: Naming does not match project guidelines.

File: deadcell/math/math.cpp [hunk 5]
```diff
@@ -200,4 +200,32 @@ void math::angle_matrix( const vec3_t &angles, matrix3x4_t &matrix ) {
...
+	if (angles > 0 && angles <= 180) {
+		angles = (angles + 180) - 360;
+	}
+	else
+	{
+		angles += 180;
+	}
+}
+
+float math::random_float(float min, float max)
```
Comment (EternityX) [inline deadcell/math/math.cpp]: This is redundant. Use ```util::misc::get_random_float_range()``` instead.

File: deadcell/math/math.cpp [hunk 6]
```diff
@@ -200,4 +200,32 @@ void math::angle_matrix( const vec3_t &angles, matrix3x4_t &matrix ) {
 	matrix[ 0 ][ 3 ] = 0.f;
 	matrix[ 1 ][ 3 ] = 0.f;
 	matrix[ 2 ][ 3 ] = 0.f;
-}
\ No newline at end of file
+}
+
+void math::angle_addition_180(float &angles) {//[-179.179]
```
Comment (EternityX) [inline deadcell/math/math.cpp]: This is already done by ```math::clamp_angles```

**Comment by TXIuTnVsbA** [comment]
= =ok,and can you use LuaBridge for Lua in csgo?

**Comment by EternityX** [comment]
> = =ok,and can you use LuaBridge for Lua in csgo?

Yes, I personally use https://github.com/ThePhD/sol2 though.

**Comment by TXIuTnVsbA** [comment]
> > = =ok,and can you use LuaBridge for Lua in csgo?
> 
> Yes, I personally use https://github.com/ThePhD/sol2 though.

thanks you !!!!!! :) ,and can you add slow walk feature, and more antiaim option,more fakelag feature?
fix it when i fight and enable the mic
plz update it

**Comment by EternityX** [comment]
> > > = =ok,and can you use LuaBridge for Lua in csgo?
> > 
> > 
> > Yes, I personally use https://github.com/ThePhD/sol2 though.
> 
> thanks you !!!!!! :) ,and can you add slow walk feature, and more antiaim option,more fakelag feature?
> fix it when i fight and enable the mic
> plz update it

You can achieve slow walk with minor modifications to the ragebot auto-stop code.
everything else will be added soon.

**Comment by TXIuTnVsbA** [comment]
> everything else will be added soon.

ok!thanks!
and,new master just snake_case ? no more change?= =i thinks its the big change for me

**Pull request closed without merging**